codegen/java/src/main/kotlin/io/spine/protodata/codegen/java/Expression.kt

```diff
@@ -0,0 +1,386 @@
+internal constructor(
+    private val message: Expression,
+    private val name: FieldName,
+    private val cardinality: CardinalityCase = SINGLE
+) {
+
+    /**
+     * A getter expression for the associated field.
+     */
+    public val getter: MethodCall
+        get() {
+            val simpleAccess = MethodCall(message, getterName)
+            return when (cardinality) {
+                LIST -> immutableListClass.call(COPY_OF, listOf(simpleAccess))
+                MAP -> immutableMapClass.call(COPY_OF, listOf(simpleAccess))
+                else -> simpleAccess
+            }
+        }
+
+    /**
+     * Constructs a setter expression for the associated field.
+     */
+    public fun setter(value: Expression): MethodCall =
+        MethodCall(message, setterName, listOf(value))
+
+    /**
+     * Constructs an `addField(..)` expression for the associated field.
+     */
+    public fun add(value: Expression): MethodCall =
+        MethodCall(message, addName, listOf(value))
+
+    /**
+     * Constructs an `addAllField(..)` expression for the associated field.
+     */
+    public fun addAll(value: Expression): MethodCall =
+        MethodCall(message, addAllName, listOf(value))
+
+    /**
+     * Constructs an `putField(..)` expression for the associated field.
+     */
+    public fun put(key: Expression, value: Expression): MethodCall =
+        MethodCall(message, putName, listOf(key, value))
+
+    /**
+     * Constructs an `putAllField(..)` expression for the associated field.
+     */
+    public fun putAll(value: Expression): MethodCall =
+        MethodCall(message, putAllName, listOf(value))
+
+    private val getterName: String
+        get() = when (cardinality) {
+            LIST -> getListName
+            MAP -> getMapName
+            else -> prefixed("get")
+        }
+
+    private val getListName: String
+        get() = "get${name.value.camelCase()}List"
+
+    private val getMapName: String
+        get() = "get${name.value.camelCase()}Map"
+
+    private val setterName: String
+        get() = when (cardinality) {
+            LIST -> addAllName
+            MAP -> putAllName
+            else -> prefixed("set")
+        }
+
+    private val addName: String
+        get() = prefixed("add")
+
+    private val addAllName: String
+        get() = prefixed("addAll")
+
+    private val putName: String
+        get() = prefixed("put")
+
+    private val putAllName: String
+        get() = prefixed("putAll")
+
+    private fun prefixed(prefix: String) =
+        "$prefix${name.value.camelCase()}"
+
+    override fun toString(): String {
+        return "FieldAccess[$message#${name.value}]"
+    }
+}
+
+/**
+ * An expression of a Java method call.
+ *
+ * Can be a static or an instance method. In the case of the former, the receiver expression
+ * is a class name. In the case of the latter — the receiver object.
+ */
+public class MethodCall
+@JvmOverloads
+constructor(
+    receiver: Expression,
```

Comment:
I still don't get what a `receiver` is. To me, `receiver` is the one who obtains the result of the expression evaluation. 

Isn't it something like `owner`, in whose context the call is performed? 

I mean, this name either requires more explanation (maybe, examples?). Or, this thing should be named differently.